torch/utils/data/dataset.py

```diff
@@ -40,6 +41,27 @@ def __add__(self, other: 'Dataset[T_co]') -> 'ConcatDataset[T_co]':
     # See NOTE [ Lack of Default `__len__` in Python Abstract Base Classes ]
     # in pytorch/torch/utils/data/sampler.py
 
+    def __getattr__(self, attribute_name):
+        if attribute_name in Dataset.functions:
+            function = functools.partial(Dataset.functions[attribute_name], self)
+            return function
+        else:
+            raise AttributeError
+
+    @classmethod
+    def register_function(cls, function_name, function):
+        Dataset.functions[function_name] = function
+
+    @classmethod
+    def register_datapipe_as_function(cls, function_name, cls_to_register):
+        if function_name in Dataset.functions:
+            raise Exception("Unable to add DataPipe function name {} as it is already taken".format(function_name))
+
+        def class_function(cls, source_dp, *args, **kwargs):
+            return cls(source_dp, *args, **kwargs)
+        function = functools.partial(class_function, cls_to_register)
+        Dataset.functions[function_name] = function
```

Comment:
We can get rid of the duplicate code for `IterableDataset` then.

Reply:
We can not, this `Dataset. functions` have to be different from `IterableDataset.functions` unless you want carefull extract super class here and use it.

Reply:
For two `classmethod` functions, we can change `Dataset` and `IterableDataset`to make it happened, since we are using decorator of `functional_datapipe` to dispatch. Am I right?

Reply:
Yeah, that's correct. I can at least remove the two class methods using `cls`.